woocommerce/woocommerce-ios
# [Mobile Payments] [Several Readers Found] Add several readers found view controller

Partially addresses #4333 

This large PR adds a new view controller to be used when multiple card readers have been discovered. It does NOT actually connect it to any flows. That will come in a subsequent PR. To test, see instructions below on how to temporarily add it to a convenient spot in the reader discovery flow.

Changes:
- Adds `SeveralReadersFoundViewController` with dummy data and no completion handling
- Handles light and dark mode
- Handles compact (and regular) vertical and horizontal size classes and orientation changes

Concerns:
- I focused on the light mode "several readers found" iOS (iPhone, portrait) designs in Figma. I made some guesses at what would be appropriate for landscape and for iPads in terms of the overall margins for the (modal) view. @Garance91540 @adamzelinski  - your feedback is needed
- Also, I noticed (late in the process) that the dark mode designs in Figma don't look like the light mode ones. They're unexpectedly full screen. I expect that is just an oversight. I hope the (modal) light mode designs that I worked from are the true design. That said, I could use some guidance especially on colors for dark mode. The modal and the background view all blend together as it is right now. See screenshots below.

To test:
- change your Run Scheme to enable simulated readers
- change `onFoundReader` in `CardReaderConnectionController` to something like this to see the new view controller
```
    func onFoundReader() {
        guard let candidateReader = candidateReader else {
            return
        }

        guard let from = fromController else {
            return
        }

        alerts.dismiss()

        let controller = SeveralReadersFoundViewController()
        from.present(controller, animated: true)

//        alerts.foundReader(
//            from: from,
//            name: candidateReader.id,
//            connect: {
//                self.state = .connectToReader
//            },
//            continueSearch: {
//                self.skippedReaderIDs.append(candidateReader.id)
//                self.candidateReader = nil
//                self.pruneSkippedReaders()
//                self.state = .searching
//            })
    }
```
- enter settings, in-person payments, manage card readers and begin a search if necessary
- rotate the device from portrait to landscape and vice versa and ensure the margins of the view are updated
- change from light to dark mode and vice versa and ensure the button colors update appropriately

Screenshots:

iPad Light Portrait
<img src="https://user-images.githubusercontent.com/1595739/131758671-b2871ec9-b393-4913-9e1e-2148ef5a188b.PNG" width=40% />

iPad Light Landscape
<img src="https://user-images.githubusercontent.com/1595739/131758705-6130c9c8-bdfe-4808-9030-d065a8771f56.PNG" width=70% />

iPad Dark Portrait
<img src="https://user-images.githubusercontent.com/1595739/131758729-8f589a8d-94cd-4c09-9c8b-495cd172456a.PNG" width=40% />

iPad Dark Landscape
<img src="https://user-images.githubusercontent.com/1595739/131758748-db52d27b-aa53-4e59-a0b4-49006cb845e3.PNG" width=70% />

iPhone Light Portrait
<img src="https://user-images.githubusercontent.com/1595739/131758806-8c52c813-330b-49a7-baa5-461c4bd39b60.PNG" width=40% />

iPhone Light Landscape
<img src="https://user-images.githubusercontent.com/1595739/131758828-64223568-8d40-4713-a0db-f51a3d0948d8.PNG" width=70% />

iPhone Dark Portrait
<img src="https://user-images.githubusercontent.com/1595739/131758852-6f950f77-44f6-48d1-8a3b-238f11db4720.PNG" width=40% />

iPhone Dark Landscape
<img src="https://user-images.githubusercontent.com/1595739/131758879-461ed066-8bec-4b45-8a90-ff87e13c54b0.PNG" width=70% />

Update release notes:
